compiler/rustc_session/src/errors.rs

```diff
@@ -170,3 +170,62 @@ pub struct StackProtectorNotSupportedForTarget<'a> {
 pub struct SplitDebugInfoUnstablePlatform {
     pub debuginfo: SplitDebuginfo,
 }
+
+#[derive(SessionDiagnostic)]
+#[diag(session::file_is_not_writeable)]
+pub struct FileIsNotWriteable<'a> {
+    pub file: &'a std::path::Path,
+}
+
+#[derive(SessionDiagnostic)]
+#[diag(session::crate_name_does_not_match)]
+pub struct CrateNameDoesNotMatch<'a> {
+    #[primary_span]
+    pub span: Span,
+    pub s: &'a str,
+    pub name: Symbol,
+}
+
+#[derive(SessionDiagnostic)]
+#[diag(session::crate_name_invalid)]
+pub struct CrateNameInvalid<'a> {
+    pub s: &'a str,
+}
+
+pub struct CrateNameEmpty {
+    pub span: Option<Span>,
```

Comment:
Does `#[primary_span]` not work on an `Option<Span>` field?

Reply:
It works and I though it didn't 😅 

Thanks David, I have just changed it. Here is the [diff](https://github.com/rust-lang/rust/compare/fbe3f99830cb4de9b814290ab1ce03d693c587c1..329d5014b6a773d819e6cd56d5930a204bed7983).